pwiz_tools/Skyline/Alerts/MsFraggerDownloadDlg.cs

```diff
@@ -241,9 +239,15 @@ private void btnAccept_Click(object sender, EventArgs e)
 
         private void btnRequestVerificationCode_Click(object sender, EventArgs e)
         {
-            RequestVerificationCode();
+            var verificationResultHtml = RequestVerificationCode();
             lblVerificationCode.Enabled = tbVerificationCode.Enabled = true;
-            MessageDlg.Show(this, AlertsResources.MsFraggerDownloadDlg_btnRequestVerificationCode_Click_Check_your_email);
+            var checkEmailDlg =
+                new AlertDlg(AlertsResources.MsFraggerDownloadDlg_btnRequestVerificationCode_Click_Check_your_email, MessageBoxButtons.OK)

```

Comment:
You can use the "ShowAndDispose" method here:
`new AlertDlg(AlertsResources.MsFraggerDownloadDlg_btnRequestVerificationCode_Click_Check_your_email, MessageBoxButtons.OK){DetailMessage = verificationResultHtml}.ShowAndDispose(this);`

It is not necessary to set "StartPosition" on AlertDlg. It gets the "CenterParent" value from "CommonAlertDlg.resx".

Reply:
That's what I thought too but when I was testing I noticed the dialog was showing up outside the parent dialog's bounds. What could be going on there?

Reply:
Maybe it's due to using Show() instead of ShowAndDispose()? It seems to work without the StartPosition when I use ShowAndDispose.

Reply:
> Maybe it's due to using Show()

The instance method "Show" on a Form should never be used for dialogs that are supposed to be modal. That brings up a non-modal window, and has a void return type.
The instance method to show a modal dialog and get back a DialogResult is "ShowDialog" (which is what "ShowAndDispose" eventually ends up calling).
"CenterParent" only matters for modal dialogs.

Reply:
Oof, thanks for the reminder. I knew that at one point. My memory must have been overwritten with the Show overloads in MessageDlg and other Skyline forms. :) Definitely explains the CenterParent thing (which I did NOT know about non-modal dialogs).